### DISCUSSION TITLE
Make generated row classes extends a subclass of DataClass

### DISCUSSION THREAD

**Discussion by mbenyoub** [simolus3/drift#1509] [Q&A]
Hi, 
i'm working on a case with lot of tables all sharing lot of system columns, and I have to implements a layer that unsure the management those columns (columns last_write_date, is_dirty , can_sync, server_id ... ) . this is why I need a generic RowClass to help me access and handle those fields

I managed make it work by editing the extended class in generated database.g. file but those changes disappear after buil_runner .

is there any way to make the generated row classes extends from a custom subclass of DataClass. 
like using anotation @ExtendsRowClass(MyDataClass) or something else

thanks

**Comment by simolus3** [comment]
There's no such thing in drift. I think this would also be pretty challenging to implement in the general case, so I don't think its likely to happen soon.
The best alternative I see is to write custom data classes for all your tables, which allows you to introduce the class structure you want.
